src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -261,12 +266,17 @@ public DynamicObject multiply(DynamicObject string, int times,
             return allocateObjectNode.allocate(Layouts.BASIC_OBJECT.getLogicalClass(string), Layouts.STRING.build(false, false, repeated));
         }
 
+        @Specialization(guards = "times >= 0")
+        public DynamicObject multiply(DynamicObject string, long times) {
+            throw new RaiseException(coreExceptions().argumentError("'long' is too big to convert into 'int'", this));
```

Comment:
FWIW JRuby has a lot of the same String/Array limitations with some existing error messages:
https://github.com/jruby/jruby/blob/0aef857789cb119b359ca8c4b5729a1c09415e2b/core/src/main/java/org/jruby/RubyString.java#L1179